text/0088-fix-memory-limitiations-in-session-replays-access-pattern.md

```diff
@@ -283,6 +283,42 @@ Use `optimize_aggregation_in_order` to limit the number of rows we need to aggre
 
 **Questions**
 
+### 12. Aggregate Replay Event Metadata in the SDK and Set Final Flag
+
+**Proposal**
+
+The SDK can maintain a buffer of the replay metadata. Once the replay has finished the SDK will flush that buffered data on the final segment. The segment marked as "final" can then be fetched without aggregation.
+
+**Drawbacks**
+
+- Requies SDK upgrade.
+- API will need to fallback to aggregating behavior if no final segments can be found (or otherwise detect old SDK usage prior to querying).
```

Comment:
This is only true if you want to show "in progress" replays. If we exclude those then this isn't a problem.

Reply:
Old SDKs won't set the `final` attribute and will require us to aggregate.  Eventually we could drop this requirement but only after some portion of customers upgraded.